src/signatures/commitment_signature.rs

```diff
@@ -81,6 +83,28 @@ where
         challenge: &[u8],
         factory: &C,
     ) -> Result<Self, CommitmentSignatureError>
+    where
+        K: Mul<P, Output = P>,
+        for<'a> &'a K: Add<&'a K, Output = K>,
+        for<'a> &'a K: Mul<&'a K, Output = K>,
+        C: HomomorphicCommitmentFactory<P = P>,
+    {
+        Self::sign_raw(secret_a, secret_x, nonce_a, nonce_x, challenge, factory)
+    }
+
+    /// Sign the provided challenge with the value commitment's value and blinding factor. The two nonces should be
+    /// completely random and never reused - that responsibility lies with the calling function.
+    ///
+    /// WARNING: The provided secret keys and nonces are NOT bound to the challenge. This method assumes that the
+    /// challenge has been constructed such that all commitments are already included in the challenge.
+    pub fn sign_raw<C>(
```

Comment:
Why was this added? As far as I can see it's the same as `sign`

Reply:
Should we replace the `sign` function by this `sign_raw`, to convey the idea that we are making assumptions that the inputs are sensible ? Similarly to what was used in the `SchnorrSignature`, via the signature api ?

Reply:
Ah right, I think this should be in another PR because we should also add a deprecation attribute to `sign` which may involve a number of other changes.